geopandas/tests/test_op_output_types.py

```diff
@@ -75,36 +75,55 @@ def assert_object(
     assert type(result) is expected_type
 
     if expected_type == GeoDataFrame:
-        if geo_name is not None:
-            _check_metadata_gdf(result, geo_name=geo_name, crs=crs)
-        else:
-            if check_none_name:  # TODO this is awkward
-                assert result._geometry_column_name is None
-
-            if result._geometry_column_name is None:
-                msg = (
-                    "You are calling a geospatial method on the GeoDataFrame, "
-                    "but the active"
-                )
-            else:
-                msg = (
-                    "You are calling a geospatial method on the GeoDataFrame, but "
-                    r"the active geometry column \("
-                    rf"'{result._geometry_column_name}'\) is not present"
-                )
-            with pytest.raises(AttributeError, match=msg):
-                result.geometry.name  # be explicit that geometry is invalid here
+        assert geo_name is not None
+        _check_metadata_gdf(result, geo_name=geo_name, crs=crs)
+
     elif expected_type == GeoSeries:
         _check_metadata_gs(result, name=geo_name, crs=crs)
 
 
+def assert_obj_none(result, expected_type, int_geo_colname=None):
```

Comment:
For the name of the helper, something like "assert_object_without_active_geometry" might be clearer (it's not always None), but maybe a bit long.

For `int_geo_colname`, the "int" in the name is coming from "intended"? (that's not super clear to me, I think it can just be `geo_colname` ?)

Reply:
names are hard 😄, I wasn't really happy with assert_object_none or int_geo_col `int` was for internal meaning `_geometry_column_name`, but if the function has a better name then I think you're right and it can just be geo_colname